http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/ClusterHelper.scala

```diff
@@ -78,9 +78,17 @@ class ClusterHelper(dbRef: DbReference,
             IO.unit
           } else {
             for {
-              emails <- getDataprocServiceAccounts(googleProject)
-              _ <- if (emails.isEmpty) IO.raiseError(ClusterIamSetupException(imageProject))
-                   else emails.traverse(e => retryIam(imageProject, e, Set("roles/compute.imageUser")))
+              projectNumber <- IO.fromFuture(IO(googleComputeDAO.getProjectNumber(googleProject))).map(_.getOrElse(throw ClusterIamSetupException(imageProject)))
```

Comment:
minor, I think it's better to `.flatMap(_.fold(IO.pure)(_ => IO.raiseError(ClusterIamSetupException(imageProject))))` to avoid throwing the exception and only to be caught somewhere else in the app

Reply:
Thanks, that looks better